code/game/machinery/medical_kiosk.dm

```diff
@@ -0,0 +1,114 @@
+//The Medical Kiosk is designed to act as a low access alernative to  a medical analyzer, and doesn't require breaking into medical. Self Diagnose at your heart's content! ...For a fee.
+
+/obj/machinery/medical_kiosk
+	name = "medical kiosk"
+	desc = "A freestanding medical kiosk, which can provide your basic medical status. Less effective than a medical analyzer."
+	icon = 'icons/obj/machines/medical_kiosk.dmi'
+	icon_state = "kiosk"
+	layer = BELOW_OBJ_LAYER
+	density = TRUE
+	circuit = /obj/item/circuitboard/machine/medical_kiosk
+	payment_department = ACCOUNT_MED
+	var/default_price = 5  //I'm defaulting to a low price on this, but in the future I wouldn't have an issue making it more or less expensive.
+	var/scan_active = null  //Shows if the machine is being used. resets upon new viewer.
+	var/adv_scan_active = null  //Shows if the machine has upgraded functionality
+	var/datum/bank_account/account  //payer's account.
+	var/mob/living/carbon/human/H   //the person using the console in each instance.
+	var/obj/item/card/id/C    //the account of the person using the console.
+
+/obj/machinery/medical_kiosk/proc/inuse()  //Verifies that the user can use the interface, followed by showing medical information.
+  if(C.registered_account)
+    account = C.registered_account
+  else
+    say("No account detected.")  //No homeless crew.
+    return
+  if(!account.has_money(default_price))
+    say("You do not possess the funds to purchase this.")  //No jobless crew, either.
+    return
+  else
+    account.adjust_money(-default_price)
+    var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_MED)
+    if(D)
+      D.adjust_money(default_price)
+    use_power(20)
+  scan_active = 0
+  icon_state = "kiosk_active"
+  say("Thank you for your patronage!")
+  adv_scan_active = 0
+  RefreshParts()
+  return
+
+/obj/machinery/medical_kiosk/update_icon()
+	if(is_operational())
+		icon_state = "kiosk_off"
+	else
+		icon_state = "kiosk"
+
+/obj/machinery/medical_kiosk/wrench_act(mob/living/user, obj/item/I) //Allows for wrenching/unwrenching the machine.
+	..()
+	default_unfasten_wrench(user, I, time = 10)
+	return TRUE
+
+/obj/machinery/medicak_kiosk/RefreshParts()
+  var/A
+  for(var/obj/item/stock_parts/scanning_module/S in component_parts)
+    A += S.rating
+  if(A >=3)
+    adv_scan_active = 1
```

Comment:
Byond's throwing that adv_scan_active isn't defined, even through I believe I defined it on line 14. Anyone know what's wrong here?